### ISSUE TITLE
Using company / charity numbers

### ISSUE THREAD

**Issue by stevieflow** [OpenDataServices/grantnav#19]
I'm a little unclear in terms of what our plans with using these are :+1: 

- during data prep - are we using these to de-dupe?
- augmenting data - are we reconciling against companies house/charity commission, to gather more info (and storing that?)
- UI - do we plan to display this info / link to the relevant registry within the organisation page(s) #13 #14 ?

@ekoner @kindly may be able to explain more

**Comment by stevieflow** [comment]
Im pushing this into iteration 3, as it will certainly be discussed

**Comment by stevieflow** [comment]
Further to this, I can see an issue for us in cases where charities are centrally organised, but grants have been made to local "branches".  

An example is the British Legion (but also see NCT, Scouts)

recipient/360G-blf-org-219279

(NB: BIG will change this identifier to GB-CHC-219279 in their next iteration)

This is a legitimate identifier for this org - they are a central charity: http://beta.charitycommission.gov.uk/charity-details/?regid=219279&subid=0 - the legal entity to which the grant was made

Our issue in the data is that we have multiple names for the ``Recipient Org: Name`` , and use the first one we see on import.  This means that British Legion becomes Royal British Legion - Gateshead Branch, which is misleading

An overhead to consider *could* be to reconcile charity (and company) names with the relevant registry, although this means a) altering the data and b) undertaking this process

Certainly, it seems that we have an issue to grapple with

cc/ @kindly @Bjwebb @ekoner @caprenter

**Comment by stevieflow** [comment]
Also - we have examples where it "looks fine", as in 

Centralised examples

- Barnaodo's: recipient/360G-blf-org-216250 (http://beta.charitycommission.gov.uk/charity-details/?regid=216250&subid=0)
- Children's Society

Distributed / different entities

- search for 'YMCA'

**Comment by ekoner** [comment]
This is a global issue but for grantNav the way forward IMO is to identify what GrantNav users wish to see and why.

There are so many facets possible with this info, it would be far too much overhead to offer them all. For instance: "Who was awarded a grant?" is a loaded and broad question.

When we say who, do we mean the org? the branch? the regional body? the school or perhaps one it's associated org? the parent charity or a sub charity? 

Knowing why the user wants that information will allow us to pick the right granularity and provide a reasonable caveat to potential users.

As for matching, my usual stance is where multiple and complex hierarchies exist, match to the most granular level as you can more easily navigate upstream to provide the *parent id than downstream. This does come with added overheads of course.

This really needs illustrating with examples, I'll add these soonish.

**Comment by stevieflow** [comment]
Thanks @ekoner 

I think I can report though - that as an (early) GrantNav user, and having spoken to others, the list of Recipient Orgs presented in the facet via a global search is confusing (and - I know it is wrong):

> Royal British Legion (Upminster and Cranham) (262)
Merched y Wawr (113)
Barnardo's (81)
Altrincham Children's Choir (76)
1st Wingate Guide Unit (72)
The Children's Society (72)
Alzheimer's Society (71)
Frome & District Scout Council (64)
National Childbirth Trust - Ascot and District Branch (61)
Volunteering Matters (60)

I do think trying to resolve this could be a near thing for us

**Comment by ekoner** [comment]
**An overview of identifiers – as I understand them**
 
![Organisation Types](https://www.dropbox.com/s/y94xyrmlproo5cx/organisation%20and%20child%20organisations.png?raw=1)

An organisation could be just a parent organisation, or it could have one or more sub-organisations as shown.

**Subsidiaries** in England & Wales typically share a charity number, even though the linked charities may have different names. This isn’t a problem in Scotland where they have completely different number

Royal British Legion (GB-CHC-219279) for example has several subsidiaries who typically provide the charity number 219279 instead of, for example

- Linked charity 219279-1 BRITISH LEGION AREA HOUSES AT ROCHDALE

**Branches** also share a charity number but will have different names, for example

- Royal British Legion (Upminster and Cranham)

**Child** is my loose term for organisations that are linked but not through any registers. They are the least likely to cause an issue as they use their own identifiers. For example, Leeds University (GB-COH-RC000658) has associated companies like:

- Bright Beginnings Childcare Centre Leeds - 06589495
- Leeds Innovation Centre Ltd - 03747084

**What does this mean for gratNav?**
Bottom line is grantNav can’t rely on just the Recipient Identifier. 

Some suggestions are:

1.     Combine name and identifier - less matches where name spellings differ (even slightly) as grantNav uses Recipient Identifier + Recipient Name (or some derivative, perhaps an auto-generated code?)
2.     Do some work upfront to match recipients to an existing or generated id that distinguishes between branches and subsidiaries.
3.     Something else I haven’t considered

**Comment by robredpath** [comment]
Closing, as this whole issue is bigger than a single GitHub issue

@ekoner is doing some great work with identifiers and 360 in general - fixing the problem further up the chain. 

In the short term, we're planning to work #167 to address the most obvious cases where this is a problem
